crates/oxc_minifier/src/peephole/replace_known_methods.rs

```diff
@@ -545,28 +559,47 @@ impl<'a> PeepholeOptimizations {
             .iter()
             .map(|arg| arg.as_expression().map(|e| e.get_side_free_number_value(ctx))?)
             .collect::<Option<Vec<_>>>()?;
-        let result = if numbers.iter().any(|n| n.is_nan()) {
-            f64::NAN
-        } else {
-            match name {
-                // TODO
-                // see <https://github.com/rust-lang/rust/issues/83984>, we can't use `min` and `max` here due to inconsistency
-                "min" => numbers.iter().copied().fold(f64::INFINITY, |a, b| {
-                    if a < b || ((a == 0f64) && (b == 0f64) && (a.to_bits() > b.to_bits())) {
-                        a
-                    } else {
-                        b
-                    }
-                }),
-                "max" => numbers.iter().copied().fold(f64::NEG_INFINITY, |a, b| {
-                    if a > b || ((a == 0f64) && (b == 0f64) && (a.to_bits() < b.to_bits())) {
-                        a
-                    } else {
-                        b
+        let result = match name {
+            "min" | "max" => {
+                if numbers.iter().any(|n| n.is_nan()) {
+                    f64::NAN
+                } else {
+                    match name {
+                        // TODO
+                        // see <https://github.com/rust-lang/rust/issues/83984>, we can't use `min` and `max` here due to inconsistency
+                        "min" => numbers.iter().copied().fold(f64::INFINITY, |a, b| {
+                            if a < b || ((a == 0f64) && (b == 0f64) && (a.to_bits() > b.to_bits()))
+                            {
+                                a
+                            } else {
+                                b
+                            }
+                        }),
+                        "max" => numbers.iter().copied().fold(f64::NEG_INFINITY, |a, b| {
+                            if a > b || ((a == 0f64) && (b == 0f64) && (a.to_bits() < b.to_bits()))
+                            {
+                                a
+                            } else {
+                                b
+                            }
+                        }),
+                        _ => unreachable!(),
                     }
-                }),
-                _ => return None,
+                }
             }
+            "imul" => {
+                if numbers.is_empty() {
+                    return None;
+                }
+                if numbers.iter().take(2).any(|n| n.is_nan() || n.is_infinite()) {
+                    0f64
+                } else {
+                    let a = numbers.first().copied()?;
```

Comment:
```suggestion
                    let a = numbers.first().copied().unwrap_or(f64::NAN);
```
Maybe we can simply this?